gtsfm/utils/geometry_comparisons.py

```diff
@@ -62,13 +62,24 @@ def align_poses_sim3(aTi_list: List[Pose3], bTi_list: List[Pose3]) -> List[Pose3
 
     aSb = Similarity3.Align(ab_pairs)
 
+    if np.isnan(aSb.scale()) or aSb.scale() == 0:
+        # TODO: handle it in GTSAM
+
+        # apply rotation and get the translation of the centroid, as we have a panorama case.
```

Comment:
Thanks for all the updates, Ayush. Do you mind adding a quick note explaining why we need the 2 separate Sim(3)'s?

Reply:
better?